File: elixir/nth-prime/nth_prime.exs
defmodule Prime do

  @doc """
  Generates the nth prime.
  """
  @spec nth(non_neg_integer) :: non_neg_integer
  def nth(count) when count < 1, do: raise ArgumentError
  def nth(count) when count < 1_000 do
    get_primes(count, 0, [])
  end
  def nth(_), do: "Stop it"

  defp get_primes(count, check, acc) do
    acc = case prime?(check) do
      true -> acc ++ [check]
      _    -> acc
    end
    if Enum.count(acc) == count do
      check
    else
      get_primes(count, check + 1, acc)
    end
  end

  # Naive prime checker: https://en.wikipedia.org/wiki/Primality_test
  defp prime?(n) do
    cond do
      n <= 1         -> false
      n <= 3         -> true
      rem(n, 2) == 0 -> false
      rem(n, 3) == 0 -> false
      true           -> prime?(n, 5)
    end
  end

  defp prime?(n, i) do
    cond do
      i * i > n          -> true
      rem(n, i) == 0     -> false
      rem(n, i + 2) == 0 -> false
      true               -> prime?(n, i + 6)
    end
  end
end

File: javascript/nth-prime/nth-prime.js
const Prime = function() {
};

Prime.nth = function(count) {
  if (count < 1) { throw new Error('Prime is not possible'); }
  let check = 0;
  let acc = [];
  while (true) {
    if (Prime.checker(check)) { acc.push(check); }
    if (acc.length == count) {
      return check;
    }
    check++;
  }
};

Prime.checker = function(n) {
  if (n <= 1) { return false; }
  if (n <= 3) { return true; }
  if (n % 2 == 0 || n % 3 == 0) { return false; }
  return Prime.checker2(n, 5);
};

Prime.checker2 = function(n, i) {
  if (i * i > n) { return true; }
  if (n % i == 0) { return false; }
  if (n % (i + 2) == 0) { return  false; }
  return Prime.checker2(n, i + 6);
};

module.exports = Prime;
